src/west/manifest.py

```diff
@@ -48,9 +54,229 @@
 # it has the exact same value as west.version.__version__ when the
 # next release is cut.
 
+#
+# Internal helpers
+#
+
+# Logging
+
 _logger = logging.getLogger(__name__)
 
-def manifest_path():
+# Manifest locating, parsing, loading, etc.
+
+_defaults = collections.namedtuple('_defaults', 'remote revision')
+
+_DEFAULT_REV = 'master'
+_YML_EXTS = ['yml', 'yaml']
+_WEST_YML = 'west.yml'
+_SCHEMA_PATH = os.path.join(os.path.dirname(__file__), "manifest-schema.yml")
+_SCHEMA_VER = parse_version(SCHEMA_VERSION)
+_EARLIEST_VER_STR = '0.6.99'  # we introduced the version feature after 0.6
+_EARLIEST_VER = parse_version(_EARLIEST_VER_STR)
+
+PathType = Union[str, os.PathLike]
+
+def _is_yml(path: PathType) -> bool:
+    return os.path.splitext(str(path))[1][1:] in _YML_EXTS
+
+def _load(data: str) -> Any:
+    try:
+        return yaml.safe_load(data)
+    except yaml.scanner.ScannerError as e:
+        raise MalformedManifest(data) from e
+
+def _west_commands_list(west_commands: Optional[Union[str, List[str]]]) -> \
+        List[str]:
+    # Convert the raw data from a manifest file to a list of
+    # west_commands locations. (If it's already a list, make a
+    # defensive copy.)
+
+    if west_commands is None:
+        return []
+    elif isinstance(west_commands, str):
+        return [west_commands]
+    else:
+        return list(west_commands)
+
+def _west_commands_maybe_delist(west_commands: List[str]) \
+        -> Union[str, List[str]]:
+    # Convert a west_commands list to a string if there's
+    # just one element, otherwise return the list itself.
+
+    if len(west_commands) == 1:
+        return west_commands[0]
+    else:
+        return west_commands
+
+def _west_commands_merge(wc1: List[str], wc2: List[str]) -> List[str]:
+    # Merge two west_commands lists, filtering out duplicates.
+
+    if wc1 and wc2:
+        return wc1 + [wc for wc in wc2 if wc not in wc1]
+    else:
+        return wc1 or wc2
```

Comment:
This function may be reduced to:

```python
return wc1 + list(set(wc2) - set(wc1)))
```

Reply:
Hmm, are you sure the set() calls and set difference operation preserve order in the same way the above does?

Reply:
Was not aware that order is relevant, should be tested.